### ISSUE TITLE
Support for multidimensional array types

### ISSUE THREAD

**Issue by vobence** [typegoose/typegoose#65]
Hi, thanks for maintaining this amazing project.

We ran into a problem during porting our Mongoose schemas to Typegoose definitions. We use multidimensional arrays in our schemas, but I haven't found any information about this use-case in the documentation. Is it supported by the lib?

E.g. I have a class definition like this:
```typescript
class Foo {
  @arrayProp({
    items: [Number]
  })
  public bar: number[][];
}

const FooModel = getModelForClass(Foo);
```
When I run the code it gives me an error:
`Error: "Foo.bar"'s Type is invalid! Type is: "function Number() { [native code] }"`

As far as I know this is a valid Mongoose schema:
```typescript
new mongoose.Schema({
  bar: [[Number]]
})
```

 Could you provide some information about this problem?
Thanks

**Comment by RyannGalea** [comment]
If it's just saving a multidimensional array, and not an array of multidimensional arrays than you can simply use I believe.
```
@prop()
bar: number[][][]
```

**Comment by hasezoey** [comment]
multidimensional arrays are not official supported (means we dont have tests for it)

and i dont really know how to support them explicitly, currently (by ryanns example) it can just be the type, which would translate to an (any) `Array`
-> any pr's for this would be welcome, but i dont currently how to implement it without fully reworking prop again

---

@RyannGalea just to make sure, in your example you used `@prop` to make an array, does it not work with `@arrayProp`?

**Comment by RyannGalea** [comment]
Currently, I am using this for my GeoJson polygon type without any issues. The @field is to do with 'type-graphql' which is also workaround by using a custom scalar type.
```
  @Field(type => TripleNumberArrayScalar)
  @prop({ required: true })
  coordinates: number[][][]
```

**Comment by hasezoey** [comment]
@RyannGalea i just mean does `coordinates: number[][][]` not "compile" to an `Array`(any) and allow everything?

**Comment by RyannGalea** [comment]
It perhaps does I am not certain, in my case because I'm using Graphql I am strictly typed to the Graphql schema anyway so any data I send is type-checked before it hits my mongoose schema.

**Comment by hasezoey** [comment]
Added this to the 7.0 Roadmap, with the following proposal:

add an option `dimensions` (or `dim`) which would be of type `Number`, this would add support for typed-multi-dimension arrays

```ts
class SomeClass {
  @arrayProp({ items: String, dimensions: 5 })
  public lengthyArray: string[][][][][]; // i hope the type is right
}
```
implementation proposal:
loop over the number of "dimensions" and at the end, add the type that is listed

**Comment by hasezoey** [comment]
added in 69f421604881a2fd7239b7958fd49c30c6c80258, shipped with 7.0+ 
-> testing would be appreciated

**Comment by hasezoey** [comment]
closing because feature is implemented since 69f421604881a2fd7239b7958fd49c30c6c80258
